### ISSUE TITLE
Change `eltype`

### ISSUE THREAD

**Issue by dpsanders** [JuliaDiff/TaylorSeries.jl#198]
The current definition of `eltype` causes problems with broadcasting.

**Comment by lbenet** [comment]
Can you be more specific, I mean, what is the problem, and which solution you suggest?

**Comment by dpsanders** [comment]
```
julia> using TaylorSeries

julia> t = Taylor1(10)
 1.0 t + 𝒪(t¹¹)

julia> t .* [1]
ERROR: MethodError: no method matching Float64(::Taylor1{Float64})
Closest candidates are:
  Float64(::T<:Number) where T<:Number at boot.jl:741
  Float64(::Int8) at float.jl:60
  Float64(::Int16) at float.jl:60
  ...
Stacktrace:
 [1] convert(::Type{Float64}, ::Taylor1{Float64}) at ./number.jl:7
 [2] setindex!(::Array{Float64,1}, ::Taylor1{Float64}, ::Int64) at ./array.jl:767
 [3] macro expansion at ./broadcast.jl:843 [inlined]
 [4] macro expansion at ./simdloop.jl:73 [inlined]
 [5] copyto! at ./broadcast.jl:842 [inlined]
 [6] copyto! at ./broadcast.jl:797 [inlined]
 [7] copy at ./broadcast.jl:773 [inlined]
 [8] materialize(::Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{1},Nothing,typeof(*),Tuple{Taylor1{Float64},Array{Int64,1}}}) at ./broadcast.jl:753
 [9] top-level scope at none:0

julia> @eval TaylorSeries eltype(t::T) where {T <: Taylor1} = T
eltype (generic function with 60 methods)

julia> t .* [1]
1-element Array{Taylor1{Float64},1}:
  1.0 t + 𝒪(t¹¹)
```

**Comment by dpsanders** [comment]
The problem (reported by @jessebett) is that broadcasting does not work correctly.
Fixing the definition of eltype as above fixes the issue.

**Comment by lbenet** [comment]
Indeed, the problem is related to our definition of `eltype`. I am not sure if this proposal is the best solution, since we use the current `eltype` definition in the actual implementation. Have you tested this? In #207 I've come with a different solution, but it certainly would be lighter to simply change the definition of `eltype`.

**Comment by lbenet** [comment]
Closing, since this is solved by #207
